modules/dnn/src/layers/tile_layer.cpp

```diff
@@ -79,18 +85,26 @@ class TileLayerImpl CV_FINAL : public TileLayer
         MatShape out_shape = shape(out);
         int rep_i, ndims = data.dims;
         int dims = 1;
-        for (int i = 0; i < ndims; i++)
-        {
-            rep_i = repeats[i];
-            if (rep_i != 1)
+        if (ndims > 1){
+            for (int i = 0; i < ndims; i++)
             {
+                rep_i = repeats[i];
+                if (rep_i != 1)
+                {
+                    tmp = tmp.reshape(0, dims);
+                    tmp = cv::repeat(tmp, 1, rep_i);
+                }
+                dims *= out_shape[i];
+            }
+            tmp = tmp.reshape(0, out_shape);
+        } else {
+            for (int i = 0; i < repeats.size(); i++){
                 tmp = tmp.reshape(0, dims);
-                tmp = cv::repeat(tmp, 1, rep_i);
+                tmp = cv::repeat(tmp, repeats[i], 1);
+                dims *= out_shape[i];
             }
-            dims *= out_shape[i];
+            tmp = tmp.reshape(0, out_shape);
```

Comment:
Do these changes imply we need to do something to `cv::repeat` so that it can be used in the same way regardless 1D or 2D mat?

Reply:
may be. @dkurt what do you think?